src/lustre/lustreParser.mly

```diff
@@ -828,9 +827,9 @@ pexpr(Q):
   (* An array constructor (not quantified) *)
   | e1 = pexpr(Q); CARET; e2 = expr { A.ArrayConstr (mk_pos $startpos, e1, e2) }
 
-  (* An array slice or tuple projection (not quantified) *)
-  | e = pexpr(Q); DOTPERCENT; i = expr 
-    { A.TupleProject (mk_pos $startpos, e, i) }
+  (* Tuple projection (not quantified) *)
+  | e = pexpr(Q); DOTPERCENT; i = NUMERAL 
+    { A.TupleProject (mk_pos $startpos, e, (int_of_string i)) }
```

Comment:
The call to `int_of_string` may raise `Failure "int_of_string"` if if the integer represented exceeds the range of integers representable in type `int`. You should handle the exception and convert it to a proper parsing error. Alternatively, you can use a Numeral instead of a `int`.

Reply:
I see that there is [another instance](https://github.com/kind2-mc/kind2/blob/develop/src/lustre/lustreParser.mly#L1063) where we use `int_of_string`. 

Do you think we will have more than 2^62 sized tuples :)

Reply:
> I see that there is another instance where we use int_of_string.

Good. Then you can fix that instance too.

> Do you think we will have more than 2^62 sized tuples :)

No, I don’t think so. That is why I suggested two alternatives instead of asking you to use a Numeral :)
However, I do think it is possible a scenario where somebody mistakenly uses a big number to index a tuple (wrong constant, bad transcompilation,...) and I would like Kind 2 to return an informative message instead of `Failure "int_of_string"`. At the bare minimum we should use a code like this one:

```
{ let idx =
    try int_of_string i 
    with _ ->
      LustreContext.fail_at_position
        (mk_pos $startpos(i)) "Tuple projection index exceeds int range"
  in
  A.TupleProject (mk_pos $startpos, e, idx) 
}

```

Reply:
sounds good. 
Fixed.